howso/engine/trainee.py

```diff
@@ -341,86 +349,79 @@ def metadata(self) -> Optional[Dict[str, Any]]:
         return deepcopy(self._metadata)
 
     @property
-    def calculated_matrices(self) -> Optional[Dict[str, DataFrame]]:
+    def calculated_matrices(self) -> None | Dict[str, DataFrame]:
```

Comment:
Feels weird to me that it would have `None` at the beginning. I feel like it should be at the end as that would be a more degenerate case. (throughout)

Reply:
`dict of str -> DataFrame | None`

`Dict[str, DataFrame] | None`